sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/PollOperation.java

```diff
@@ -45,35 +46,14 @@
         return context -> {
             PollingState pollingState = PollingState.from(serializerAdapter, context);
             if (pollingState.getOperationStatus().isComplete()) {
-                if (pollingState.getOperationStatus() == LongRunningOperationStatus.FAILED
-                    || pollingState.getOperationStatus() == LRO_CANCELLED) {
-                    // Failed|Cancelled
-                    Error lroInitError = pollingState.getSynchronouslyFailedLroError();
-                    if (lroInitError != null) {
-                        return errorPollResponseMono(pollingState.getOperationStatus(), lroInitError);
-                    }
-                    Error pollError = pollingState.getPollError();
-                    if (pollError != null) {
-                        return errorPollResponseMono(pollingState.getOperationStatus(), pollError);
-                    }
-                    throw new IllegalStateException("Either LroError or PollError must"
-                        + "be set when OperationStatus is in Failed|Cancelled State.");
-                } else {
-                    // Succeeded
-                    return pollResponseMono(serializerAdapter,
-                        pollingState.getOperationStatus(),
-                        pollingState.getLastResponseBody(),
-                        pollResultType);
-                }
+                return pollResponseMonoFromPollingState(serializerAdapter, pollResultType, pollingState);
             } else {
                 // InProgress|NonTerminal-Status
-                Mono<PollResponse<PollResult<T>>> pollResponse = pollResponseMono(serializerAdapter,
-                    pollingState.getOperationStatus(),
-                    pollingState.getLastResponseBody(),
-                    pollResultType);
                 return doSinglePoll(pipeline, pollingState)
-                    .doOnNext(updatedState -> updatedState.store(context))
-                    .then(pollResponse);
```

Comment:
Here is the bug. Previously, after `doSinglePoll`, the previous `pollResponse` is returned. Should use the latest after `doSinglePoll`.